packages/components/src/mobile/keyboard-avoiding-view/index.ios.js

```diff
@@ -40,8 +40,13 @@ export const KeyboardAvoidingView = ( {
 
 	const animatedHeight = useRef( new Animated.Value( MIN_HEIGHT ) ).current;
 
+	const isLandscape = () => {
+		const dim = Dimensions.get( 'screen' );
+		return dim.width >= dim.height;
+	};
+
 	const { height: fullHeight } = Dimensions.get( 'window' );
-	const keyboardVerticalOffset = fullHeight - parentHeight;
+	const keyboardVerticalOffset = isLandscape ? 0 : fullHeight - parentHeight;
```

Comment:
Hey @antonis 👋 I've been testing this on the simulator and looks like `Dimensions.get( 'window' )` is returning the wrong width/height values while switching between `landscape` and `portrait`.

I did some tests by reverting these changes and just changing:

`Dimensions.get( 'window' );`

To:

`Dimensions.get( 'screen' );`

And it seems to fix the issue since it calculates the right offset. Have you tried that approach?

Reply:
Hey @geriux 👋 
Thank you for testing this. I tried it at some point but was a bit skeptical on if I should use it since the observed behaviour seemed a bit unpredictable too me. Since the added logic on detecting the screen orientation is based on the `Dimensions.get( 'screen' )` anyway, I think your suggestion makes perfect sense. Updated with 949d62c35eb21e51d474ba65caaebb07440f4c56

Reply:
> It's also interesting that Dimensions.get( 'screen' ) and the docs are not so clear on the differences 🤔

Looks like the [only difference](https://github.com/facebook/react-native/blob/f48b54bf6211dbbdd32839b0de68b1d0a451e486/Libraries/Utilities/Dimensions.js#L47) is in Android

 It does make me wonder why those values aren't being updated for `window`. We could move forward with this or take it even further and add a `change` listener like [this](https://reactnative.dev/docs/dimensions#example), to make sure we always have the updated size. What do you think?

Reply:
As discussed, since the listener with `window` is still causing issues we'll leave it as it is.